src/diagram/components/dialog/edit-variable-dialog.tsx

```diff
@@ -1,74 +1,127 @@
 import React, { useMemo } from "react";
 import { observer } from "mobx-react-lite";
-import { applySnapshot, getSnapshot } from "mobx-state-tree";
-
+import { applySnapshot, getSnapshot, IAnyComplexType, Instance } from "mobx-state-tree";
 import { NumberRow } from "./number-row";
 import { TextRow } from "./text-row";
 import { TextAreaRow } from "./text-area-row";
 import { Variable, VariableType } from "../../models/variable";
 import { kMaxNameCharacters, kMaxNotesCharacters, processName } from "../../utils/validate";
+import classNames from "classnames";
+import { IconInputCard } from "./icon-input-card";
+import { IconOutputCard } from "./icon-output-card";
+import { ErrorMessage } from "../error-message";
 
 import "./dialog.scss";
 
-interface IEditVariableDialogContent {
-  variable: VariableType;
+export interface IEditVariableDialogContent {
+  variable?: VariableType;
+  variableClone: VariableType;
 }
-export const EditVariableDialogContent = observer(({ variable }: IEditVariableDialogContent) => {
+
+export const EditVariableDialogContent = observer(({ variable, variableClone }: IEditVariableDialogContent) => {
+  // Because variableClone was created from a snapshot, we will not be able to access 
+  // computed values related to the original variable's inputs. We need to get those values
+  // from the original variable.
+  const errorMessage = variable?.computedValueError || variable?.computedUnitError || variable?.computedUnitMessage;
+  const isExpressionVariable = variable?.hasInputs;
+
   const updateName = (newName: string) => {
-    variable.setName(processName(newName));
+    variableClone.setName(processName(newName));
+  };
+  const updateValue = (newValue: number | undefined) => {
+    variableClone.setValue(newValue);
   };
```

Comment:
Would it be possible to directly pass `variableClone.setValue` as the `setRealValue` prop below, instead of wrapping it in `updateValue`? I know this was to fix an issue with testing, but I think it's worth another shot, because I don't think the wrapper is actually doing anything.

Reply:
Yes, thanks for pointing that out. I've made that change.